api/controllers/MerchStoreController.ts

```diff
@@ -181,15 +187,32 @@ export class MerchStoreController {
     return { error: null, order };
   }
 
-  @Patch('/order')
-  async editMerchOrder(@Body() fulfillOrderRequest: FulfillMerchOrderRequest,
+  @Patch('/order/:uuid')
+  async editMerchOrder(@Params() params: UuidParam,
+    @Body() editOrderRequest: EditMerchOrderRequest,
     @AuthenticatedUser() user: UserModel): Promise<EditMerchOrderResponse> {
+    if (!PermissionsService.canAccessMerchStore(user)) throw new ForbiddenError();
+    const { status, pickupEvent } = editOrderRequest.order;
+    // status and pickup event are not allowed to be updated at the same time,
+    // since they are not logically related, and thus should never have to be updated at the same request
+    if (status && pickupEvent) throw new UserError('Only status or pickupEvent can be updated at once, not both');
+    // members are only allowed to cancel orders. store admins can perform any operation
+    if (status && status !== OrderStatus.CANCELLED && !PermissionsService.canEditMerchStore(user)) {
+      throw new ForbiddenError('Members can only cancel orders');
+    }
+    await this.merchStoreService.editOrder(params.uuid, editOrderRequest.order, user);
+    return { error: null };
+  }
+
+  @Post('/order/:uuid/fulfillment')
```

Comment:
I'm a fan of it too - personally I'm of the opinion of the 'convention' for REST API naming is whatever naming we establish (since I don't think there's a set of 'naming conventions' for APIs). So I don't think we should feel too bad about having verbs in the name - especially since it also distinguishes 'actions' (ie. reset password or fulfill orders) from 'resource operations' (ie. editing a field of a model).